src/vkcapture.c

```diff
@@ -635,7 +637,7 @@ static obs_properties_t *vkcapture_source_get_properties(void *data)
         obs_property_list_add_string(p, name, value);
     }
 
-    if (cursor_enabled(ctx)) {
+    if (ctx == NULL || cursor_enabled(ctx)) {
```

Comment:
`if (ctx && cursor_enabled(ctx))`

Reply:
I add the property on purpose if the `ctx` is null, because when invoked with a null pointer it's used to create the possible UI elements which can then be used by other plugins to find out what may be present in the config dialog.

If we require a valid context, then there is no way of finding out about this extra boolean field in the settings.

Reply:
Alright, but please change it to `if (!ctx || cursor_enabled(ctx))`

Any example of a plugin that would use this?

Reply:
No concrete example as of now. I discovered this while creating bindings to libobs and tried out various APIs. 

It can be helpful in tandem with the default settings as it can provide further details about valid value ranges (number sliders for example).

Turns out, most plugins don't expect a null and use it without check, which causes OBS to crash with a SEGFAULT. Even OBS maintainers weren't aware of this oversight in the API, so several built-in plugins crash due to this as well (a few were fixed after I raised the issue).